src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int OFFSET = 1;
+    private static final int START_INDEX = 0;
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double ELEMENT_DATA_MULTIPLIER = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < START_INDEX) {
+            checkIndex(index);
```

Comment:
```suggestion
        checkIndexForAdd(index);
```

create separate method